x/stakeibc/keeper/unbonding_records.go

```diff
@@ -500,7 +583,10 @@ func (k Keeper) InitiateAllHostZoneUnbondings(ctx sdk.Context, dayNumber uint64)
 		}
 
 		// Get host zone unbonding message by summing up the unbonding records
-		if err := k.UnbondFromHostZone(ctx, hostZone); err != nil {
+		err := utils.ApplyFuncIfNoError(ctx, func(ctx sdk.Context) error {
```

Comment:
I think this is ok, but double checking because it's different than what we discussed live - this makes (1) updating the records and (2) sending the MsgUndelegate ICAs atomic, right?

Reply:
nit: should we rename to `UpdateRecordsAndSendUnbondingICAs`?

Reply:
Ah sorry - must have misunderstood. I though we wanted both (1) and (2) to be atomic, with the thinking being that there's no need to do (1) outside of the context of (2)?

Reply:
Wasn't making them atomic the plan we discussed live?

> Only execute (2) if (1) succeeds

Reply:
re: naming, my thinking was we have a few core operations in stakeibc (delegation, unbonding, redemptionSweep, etc.). So I thought it made sense to name them using just that high level category as the name instead of getting too specific (e.g. `UnbondFromHostZone`, `DelegateToHostZone`, etc.). Makes it a little easier to locate the main function imo instead of conflating it with helpers

That said, open to something else, but that was my thinking!

Reply:
> Wasn't making them atomic the plan we discussed live?

I think we discussed not rolling back setting native token amounts if sending ICAs failed, but might have misunderstood. Either way, not important and I'm signed off on this approach 

> re: naming...

I bias to explicit clarity in naming variables and functions (at the expense of making them verbose), but don't feel strongly

Reply:
let's table naming for now, still planning on doing the file org refactor soon we can re-evaluate then